criar.sql

```diff
@@ -34,37 +33,37 @@ CREATE TABLE Document (
 	date_fin	DATE,
 	type		TEXT,
 	ownerID		INTEGER,
-	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID) ON DELETE RESTRICT,
-	CONSTRAINT document_owner_fk FOREIGN KEY (ownerID) REFERENCES Worker(WID) ON DELETE SET NULL,
-	CONSTRAINT checkDiDf CHECK(date_ini<date_fin OR date_fin=NULL)
+	CONSTRAINT document_type_fk FOREIGN KEY (type) REFERENCES Type(TID) ON DELETE RESTRICT ON UPDATE CASCADE,
+	CONSTRAINT document_owner_fk FOREIGN KEY (ownerID) REFERENCES Worker(WID) ON DELETE SET NULL ON UPDATE CASCADE,
+	CONSTRAINT checkDiDf CHECK (date_ini<date_fin OR date_fin=NULL)
 );
 CREATE TABLE Language (
 	LID			INTEGER,
-	Lname		TEXT NOT NULL,
+	Lname		TEXT UNIQUE NOT NULL,
 	CONSTRAINT language_pk PRIMARY KEY (LID)
 );
 CREATE TABLE File (
 	FID			INTEGER,
 	docID		INTEGER NOT NULL,
-	Fname		TEXT,
+	Fname		TEXT UNIQUE NOT NULL,
 	size		INTEGER,
 	extension	TEXT,
 	CONSTRAINT file_pk PRIMARY KEY (FID), 
-	CONSTRAINT file_doc_fk FOREIGN KEY (docID) REFERENCES Document(docID) ON DELETE CASCADE,
-	CONSTRAINT checkFSize CHECK(size >=0)
+	CONSTRAINT file_doc_fk FOREIGN KEY (docID) REFERENCES Document(docID) ON DELETE CASCADE ON UPDATE CASCADE,
+	CONSTRAINT checkFSize CHECK (size >=0)
 );
 CREATE TABLE DocAttribute (
 	DAID		INTEGER,
-	DAname		TEXT NOT NULL,
-	DAvalue		TEXT,
+	DAname		TEXT UNIQUE NOT NULL,
+	DAvalue		TEXT UNIQUE NOT NULL,
 	CONSTRAINT docAttribute_pk PRIMARY KEY (DAID) 
 );
 CREATE TABLE Type (
 	TID			INTEGER,
 	Tname		TEXT NOT NULL,
 	last_mod_date	DATE,
-	status		TEXT,
-	remarks		TEXT,
+	status		INTEGER NOT NULL,
+	remarks		TEXT UNIQUE NOT NULL,

```

Comment:
Mudei o status para INTEGER e NOT NULL porque ou está ativo ou inativo, não há nenhum outro estado.
Mudei as remarks para unique not null porque um tipo tem que ter descrição e dois tipos com a mesma descrição têm que ser o mesmo tipo.

Reply:
```suggestion
	status		INTEGER NOT NULL,
	remarks		TEXT UNIQUE NOT NULL,
	CONSTRAINT status_bool CHECK (status=0 OR status=1)
```
Lembrei-me que falta uma check constraint para ver se o status é mesmo só 0 ou 1.